dev/tools/lib/roll_dev.dart

```diff
@@ -0,0 +1,158 @@
+const String kHelp = 'help';
+
+void main(List<String> args) {
+  // If we're run from the `tools` dir, set the cwd to the repo root.
+  if (path.basename(Directory.current.path) == 'tools')
+    Directory.current = Directory.current.parent.parent;
+
+  final ArgParser argParser = new ArgParser(allowTrailingOptions: false);
+  argParser.addOption(
+    kIncrement,
+    help: 'Specifies which part of the x.y.z version number to increment. Required.',
+    valueHelp: 'level',
+    allowed: <String>[kX, kY, kZ],
+    allowedHelp: <String, String>{
+      kX: 'Indicates a major development, e.g. typically changed after a big press event.',
+      kY: 'Indicates a minor development, e.g. typically changed after a beta release.',
+      kZ: 'Indicates the least notable level of change. You normally want this.',
+    },
+  );
+  argParser.addFlag(kHelp, negatable: false, help: 'Show this help message.', hide: true);
+  ArgResults argResults;
+  try {
+    argResults = argParser.parse(args);
+  } on ArgParserException catch (error) {
+    print(error.message);
+    print(argParser.usage);
+    exit(1);
+  }
+
+  final String level = argResults[kIncrement];
+  final bool help = argResults[kHelp];
+
+  if (help || level == null) {
+    print('roll_dev.dart --increment=x • update the version tags and roll a new dev build.\n');
+    print(argParser.usage);
+    exit(0);
+  }
+
+  runGit('checkout master', 'switch to master branch');
+
+  if (getGitOutput('status --porcelain', 'check status of your local checkout') != '') {
+    print('Your git repository is not clean. Try running "git clean -fd". Warning, this ');
+    print('will delete files! Run with -n to find out which ones.');
+    exit(1);
+  }
+
+  String version = getFullTag();
+  final Match match = parseFullTag(version);
+  if (match == null) {
+    print('Could not determine the version for this build.');
+    if (version.isNotEmpty)
+      print('Git reported the latest version as "$version", which does not fit the expected pattern.');
+    exit(1);
+  }
+
+  final List<int> parts = match.groups(<int>[1, 2, 3]).map(int.parse).toList();
+
+  if (match.group(4) == '0') {
+    print('This commit has already been released, as version ${parts.join(".")}.');
+    exit(0);
+  }
+
+  switch (level) {
+    case kX:
+      parts[0] += 1;
+      parts[1] = 0;
+      parts[2] = 0;
+      break;
+    case kY:
+      parts[1] += 1;
+      parts[2] = 0;
+      break;
+    case kZ:
+      parts[2] += 1;
+      break;
+    default:
+      print('Unknown increment level. The valid values are "$kX", "$kY", and "$kZ".');
+      exit(1);
+  }
+  version = parts.join('.');
+
+  runGit('fetch upstream', 'fetch upstream');
+  runGit('reset upstream/master --hard', 'check out master branch');
+  runGit('tag $version', 'tag the commit with the version label');
+
+  print('Your tree is ready to publish Flutter $version to the "dev" channel.');
+  stdout.write('Are you? [yes/no] ');
+  if (stdin.readLineSync() != 'yes') {
```

Comment:
We may want to respect 'y' as well.

Reply:
Two extra characters per person every few months is probably a cost worth paying for the extra level of confidence in the confirmation. :-P